lightning/src/routing/gossip.rs

```diff
@@ -993,11 +1090,51 @@ impl fmt::Display for NodeInfo {
 	}
 }
 
-impl_writeable_tlv_based!(NodeInfo, {
-	(0, lowest_inbound_channel_fees, option),
-	(2, announcement_info, option),
-	(4, channels, vec_type),
-});
+impl Writeable for NodeInfo {
+	fn write<W: ::util::ser::Writer>(&self, writer: &mut W) -> Result<(), io::Error> {
+		write_tlv_fields!(writer, {
+			(0, self.lowest_inbound_channel_fees, option),
+			(2, self.announcement_info, option),
+			(4, self.channels, vec_type),
+		});
+		Ok(())
+	}
+}
+
+// A wrapper allowing for the optional deseralization of `NodeAnnouncementInfo`. Utilizing this is
+// necessary to maintain compatibility with previous serializations of `NodeAnnouncementInfo` that have an
+// invalid hostname set. In this case, we simply ignore the error and continue reading the `NodeInfo`.
+struct NodeAnnouncementInfoDeserWrapper(NodeAnnouncementInfo);
+
+impl MaybeReadable for NodeAnnouncementInfoDeserWrapper {
+	fn read<R: io::Read>(reader: &mut R) -> Result<Option<Self>, DecodeError> {
+		match ::util::ser::Readable::read(reader) {
+			Ok(node_announcement) => Ok(Some(Self(node_announcement))),
+			Err(DecodeError::ShortRead) => Ok(None),
```

Comment:
Not entirely sure if this would also fail with a `ShortRead` in case of an invalid hostname. I should probably double-check this.

Reply:
> Yea, I think there's no question we're gonna end up accepting tons of cases that should be actual errors, but that's okay. That said, I don't think the case we actually care about here will generate a `ShortRead`, I think "missing required TLV" is an `InvalidValue` error

No, at least in the case for `ChannelUpdateInfo` I checked: the read error returns an `ShortRead` (otherwise  tests would currently fail).

> can we add a test that explicitly tries to read some hard-coded hex value of a network graph and check that it passes?

Sure, but isn't that kind of what the `written_channel_info_is_readable` above does?

Reply:
I now check that we succeed/fail to read previously serialized `ChannelUpdateInfo`s, depending on whether the `htlc_maximum_msat` field is present or `None` (see https://github.com/lightningdevkit/rust-lightning/pull/1519/commits/331996b4ba8b709a5750ee2c8479f8b9169bb40c).